src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataForeignKeyTest.java

```diff
@@ -100,22 +95,40 @@ public static void terminateVariation() throws SQLException {
      */
     @Test
     public void testGetImportedKeys() throws SQLException {
-        SQLServerDatabaseMetaData dmd = (SQLServerDatabaseMetaData) connection.getMetaData();
-
-        SQLServerResultSet rs1 = (SQLServerResultSet) dmd.getImportedKeys(null, null, table1);
-        validateGetImportedKeysResults(rs1);
-
-        SQLServerResultSet rs2 = (SQLServerResultSet) dmd.getImportedKeys(catalog, schema, table1);
-        validateGetImportedKeysResults(rs2);
-
-        SQLServerResultSet rs3 = (SQLServerResultSet) dmd.getImportedKeys(catalog, "", table1);
-        validateGetImportedKeysResults(rs3);
-
-        try {
-            dmd.getImportedKeys("", schema, table1);
-            fail(TestResource.getResource("R_expectedExceptionNotThrown"));
-        } catch (SQLException e) {
-            assertTrue(e.getMessage().startsWith(TestResource.getResource("R_dbNameIsCurrentDB")));
+        SQLServerResultSet rs1 = null;
+        SQLServerResultSet rs2 = null;
+        SQLServerResultSet rs3 = null;
+     
+        try (SQLServerConnection conn = (SQLServerConnection) DriverManager.getConnection(connectionString)) {
+            SQLServerDatabaseMetaData dmd = (SQLServerDatabaseMetaData) conn.getMetaData();
+
+            rs1 = (SQLServerResultSet) dmd.getImportedKeys(null, null, table1);
```

Comment:
Move all ResultSets to one try w. res. block - validate method calls can happen all at once inside the block..